crux-mir/src/Mir/Language.hs

```diff
@@ -155,30 +155,33 @@ withMirLogging computation =
         ?injectMirLogMessage = LoggingMir
      in computation
 
-mainWithOutputConfig :: (Maybe Crux.OutputOptions -> OutputConfig MirLogging)
-                     -> BindExtraOverridesFn -> IO ExitCode
-mainWithOutputConfig mkOutCfg bindExtra =
+mainWithOutputConfig ::
+    (forall sym. IO (s sym)) ->
+    (Maybe Crux.OutputOptions -> OutputConfig MirLogging) ->
+    BindExtraOverridesFn s -> IO ExitCode
+mainWithOutputConfig initS mkOutCfg bindExtra =
     withMirLogging $
     Crux.loadOptions mkOutCfg "crux-mir" Paths_crux_mir.version mirConfig
-        $ runTestsWithExtraOverrides bindExtra
+        $ runTestsWithExtraOverrides initS bindExtra
 
-type BindExtraOverridesFn = forall sym bak p t st fs args ret blocks rtp a r.
+type BindExtraOverridesFn s = forall sym bak p t st fs args ret blocks rtp a r.
     (C.IsSymInterface sym, sym ~ W4.ExprBuilder t st fs) =>
+    s sym ->
     Maybe (Crux.SomeOnlineSolver sym bak) ->
     CollectionState ->
     Text ->
     C.CFG MIR blocks args ret ->
     Maybe (C.OverrideSim (p sym) sym MIR rtp a r ())
 
-noExtraOverrides :: BindExtraOverridesFn
-noExtraOverrides _ _ _ _ = Nothing
+noExtraOverrides :: BindExtraOverridesFn Maybe
```

Comment:
`noExtraOverrides` isn't making meaningful use of the user state argument, so couldn't this be generalized from `Maybe` to any state type `s`?